lib/contentProfileListener.js

```diff
@@ -0,0 +1,154 @@
+  }
+
+  get specifications() { return this._specs; }
+
+  importFile(file) {
+    // Setup a child logger to associate logs with the current file
+    const lastLogger = logger;
+    logger = rootLogger.child({ file: file });
+    logger.debug('Start importing content profile file');
+    try {
+      const errListener = new SHRErrorListener(logger);
+      const chars = new FileStream(file);
+      const lexer = new SHRContentProfileLexer(chars);
+      lexer.removeErrorListeners();
+      lexer.addErrorListener(errListener);
+      const tokens  = new CommonTokenStream(lexer);
+      const parser = new SHRContentProfileParser(tokens);
+      parser.removeErrorListeners();
+      parser.addErrorListener(errListener);
+      parser.buildParseTrees = true;
+      const tree = parser.doc();
+      const walker = new ParseTreeWalker();
+      walker.walk(this, tree);
+    } finally {
+      logger.debug('Done importing content profile file');
+      this.logger = lastLogger;
+    }
+  }
+
+  enterDoc(ctx) {
+    // set grammar version
+    const version = ctx.docHeader().version();
+    const major = parseInt(version.WHOLE_NUMBER()[0], 10);
+    const minor = parseInt(version.WHOLE_NUMBER()[1], 10);
+    this._currentGrammarVersion = new Version(major, minor);
+
+    logger.debug({version: this._currentGrammarVersion.toString()}, 'Entered content profile file');
+  }
+
+  exitDoc(ctx) {
+    // clear current namespace, current content, current rule, and grammar version
+    logger.debug('Exiting content profile file');
+    this._currentNs = null;
+    this._currentDef = null;
+    this._currentRule = null;
+    this._currentGrammarVersion = null;
+  }
+
+  enterNamespaceHeader(ctx) {
+    // set current namespace
+    this._currentNs = ctx.namespace().getText();
+  }
+
+  enterContentHeader(ctx) {
+    // set current content
+    const name = ctx.simpleName().getText();
+    const identifier = new Identifier(this._currentNs, name);
+    this._currentDef = new ContentProfile(identifier);
+    this._currentDef.grammarVersion = this._currentGrammarVersion;
+  }
+
+  enterCpRule(ctx) {
+    // find identifier for each data element in path,
+    // then create and set current rule with that path
+
+    const pathStr = ctx.simpleOrPathName().getText();
+    const names = pathStr.split('.');
+    let path = [];
+    let currentElement = this._specs.dataElements.findByIdentifier(this._currentDef.identifier);
+
+    if (pathStr === 'Value' || pathStr === '_Value') {
+      path.push(new Identifier('', '_Value'));
+    } else {
+      for (const name of names) {
+        let value = [currentElement.value, ...currentElement.fields].find(field => {
+          let identifier;
+  
+          if (!field) {
+            return;
+          }
+  
+          if (field.effectiveIdentifier) { // is IdentifiableValue
+            identifier = field.possibleIdentifiers.find(id => {
+              return (name === id.name);
+            });
```

Comment:
It's a stylistic thing, but this can be shortened to a single line:
```js
identifier = field.possibleIdentifiers.find(id => name === id.name);
```
Same for the one below (line 115-117).